### ISSUE TITLE
Classloader issues

### ISSUE THREAD

**Issue by Zbob750** [PaperMC/Paperclip#4]
There appear to be several issues with the classloading method used by Paperclip.
Most of the internal team are aware of them, despite this we are unaware of a solution at this time.

So this ticket is sort of a "master" for all of those. Perhaps its simple and overlooked, perhaps it's a limitation of the system. :shrug:

**Comment by jwulf** [comment]
The classloading method used by Paperclip seems incompatible with Java 9's new modular pattern [JEP-220](http://openjdk.java.net/jeps/220). I get the error:

```SystemClassLoader not URLClassLoader```

Running paperclip.jar with Java 9.

Any plans to support Java 9?

**Comment by Zbob750** [comment]
Not until you it's released no

**Comment by Zbob750** [comment]
Update on Java 9 support: Jigsaw, the module system, is dead. Voted down by Java's EC. This means that we will probably not need to update this for any special java 9 support.

This is why we held off, major stuff can change :)

**Comment by kashike** [comment]
![x](http://screencrush.com/442/files/2014/10/jigsaw.jpg)

**Comment by Zbob750** [comment]
Going to split this into two separate tickets. This was not intended to be about Java 9's classloader changes, but it morphed into it.
